modules/csharp/examples/code_snippets/VectorSearch.cs

```diff
@@ -0,0 +1,203 @@
+            // tag::vs - create - default - config[]
+            // Create a vector index configuration for indexing 3 dimensional vectors embedded
+            // in the documents' key named "vector" using 2 centroids.
+            var config = new VectorIndexConfiguration("vector", 3, 2);
+            // end::vs - create - default - config[]
+        }
+
+        private void CreateCustomVectorIndexConfig()
+        {
+            // tag::vs - create - default - config[]
+            // Create a vector index configuration for indexing 3 dimensional vectors embedded
+            // in the documents' key named "vector" using 2 centroids. The config is customized
+            // to use Cosise distance metric, no vector encoding, min training size 100 and
+            // max training size 200.
+            var config = new VectorIndexConfiguration("vector", 3, 2)
+            {
+                DistanceMetric = DistanceMetric.Cosine,
+                Encoding = VectorEncoding.None(),
+                MinTrainingSize = 100,
+                MaxTrainingSize = 200
+            };
+            // end::vs - create - default - config[]
+        }
+
+        private void CreateVectorIndex()
+        {
+            var database = new Database("my-database");
+
+            // tag::vs-create-index[]
+            // Create a vector index configuration for indexing 3 dimensional vectors embedded
+            // in the documents' key named "vector" using 2 centroids.
+            var config = new VectorIndexConfiguration("vector", 3, 2);
+
+            // Create a vector index named "color_index" using the configuration
+            var collection = database.GetCollection("colors");
+            collection.CreateIndex("colors_index", config);
+            // end::vs-create-index[]
+        }
+
+        // tag::vs-predictive-model[]
+        public sealed class ColorModel : IPredictiveModel
+        {
+            public DictionaryObject Predict(DictionaryObject input)
+            {
+                // Get the input color code 
+                var inputColor = input.GetString("colorInput");
+                if (inputColor == null)
+                {
+                    return null;
+                }
+
+                // Get a vector, an array of float numbers, for the input color code.
+                // Normally, you will get the vector from your ML model.
+                float[] vector;
+                try
+                {
+                    vector = Color.GetVector(inputColor);
+                }
+                catch (Exception)
+                {
+                    return null;
+                }
+
+                // Create an output dictionary by setting the vector result to
+                // the dictionary key named "vector".
+                var retVal = new MutableDictionaryObject();
+                retVal.SetValue("vector", vector);
+                return retVal;
+            }
+        }
+        // tag::end-predictive-model[]
```

Comment:
this should be `end::vs-predictive-model[]`
@ElliotFrancisHunter you'll need to go through all the tags/ends and make sure they match